ui/component/or-app/src/index.ts

```diff
@@ -283,15 +298,27 @@ export class OrApp<S extends AppStateKeyed> extends LitElement {
             return;
         }
 
-        if (changedProps.has("_page")) {
+        // If either page or 'offline'-status is changed, it should update to the correct page,
+        // by appending the page to the HTML content
+        if (changedProps.has("_page") || changedProps.has("_offline")) {
             if (this._mainElem) {
                 if (this._mainElem.firstElementChild) {
                     this._mainElem.firstElementChild.remove();
                 }
                 if (this._page) {
                     const pageProvider = this.appConfig!.pages.find((page) => page.name === this._page);
                     if (pageProvider) {
-                        const pageElem = pageProvider.pageCreator();
+                        let pageElem;
+                        if(!this._offline || pageProvider.allowOffline) {
```

Comment:
If already on a page that allows offline then it will be recreated when offline changes which is probably undesirable esp. if that page has input fields etc.